torch/testing/_internal/common_device_type.py

```diff
@@ -283,7 +283,11 @@ def instantiated_test(self, name=name, test=test_fn, dtype=dtype, op=op):
             for op in test.op_list:
                 # Acquires dtypes, using the op data if unspecified
                 dtypes = cls._get_dtypes(test)
-                if dtypes is None:
+                if dtypes is not None:
+                    want_supported = not test.unsupported_dtypes_only
+                    dtypes = [t for t in dtypes
+                              if op.supports_dtype(t, cls.device_type) == want_supported]
```

Comment:
This new behaviour is useful for #48428 because while the transforms do technically support integer inputs, they are just up-cast to float so it's not worth testing all integer types exhaustively.